### DISCUSSION TITLE
sql formatting

### DISCUSSION THREAD

**Discussion by michaelzusev** [LazyVim/LazyVim#291] [Q&A]
Hi, I'm having trouble with getting sql to be formatted. I've installed neovim and lazyvim, as well as: sqlfluff, sqlls and sql-formatter in mason, and the other language LSPs and Formatters work as expected, but when I try to use the formatting features on files ending in .sql, they don't work. Is there a way I can troubleshoot?

**Comment by folke** [comment]
Did you configure an lsp or null-ls to use those sql formatters?

**Reply by michaelzusev** [reply]
I'm sorry, I'm not sure what that means. With other languages, like typescript and python, I only installed the modules through mason and it works. Is there something I need to configure for sql files?

**Reply by michaelzusev** [reply]
Ok I think I've almost figured it out. I've installed sqlfluff on mason, and pip installed it on my computer. Then I configured a lua file in plugins like this: 

```
return {
  {
    "jose-elias-alvarez/null-ls.nvim",
    opts = function()
      local nls = require("null-ls")
      return {
        sources = {
          -- nls.builtins.formatting.prettierd,
          nls.builtins.formatting.stylua,
          nls.builtins.diagnostics.flake8,
          nls.builtins.formatting.sqlfluff,
          nls.builtins.diagnostics.sqlfluff,
        },
      }
    end,
  },
}

```

And then added a .sqlfluff file in my root project directory where dialect=bigquery

I am getting the diagnostics to show up. However, when I do space cf, I see the progress bar in the bottom right with sqlfluff, but then I get a warning: [LSP] [null-ls] timeout. The progress bar has a checkmark but nothing in the file has changed. 

I've done something like this: 

```
return {
  {
    "jose-elias-alvarez/null-ls.nvim",
    opts = function()
      local nls = require("null-ls")
      return {
        sources = {
          -- nls.builtins.formatting.prettierd,
          nls.builtins.formatting.stylua,
          nls.builtins.diagnostics.flake8,
          nls.builtins.formatting.sqlfluff,
          nls.builtins.diagnostics.sqlfluff,
        },
        default_timeout = -1,
      }
    end,
  },
} 
```

But it has the same behaviour. Any ideas?
